src/order/complete_lattice.lean

```diff
@@ -1292,6 +1292,58 @@ instance [has_Sup α] [has_Sup β] : has_Sup (α × β) :=
 instance [has_Inf α] [has_Inf β] : has_Inf (α × β) :=
 ⟨λ s, (Inf (prod.fst '' s), Inf (prod.snd '' s))⟩
 
+variables {α β}
+
+lemma fst_Inf [has_Inf α] [has_Inf β] (s : set (α × β)) :
+  (Inf s).fst = Inf (prod.fst '' s) := rfl
+
+lemma snd_Inf [has_Inf α] [has_Inf β] (s : set (α × β)) :
+  (Inf s).snd = Inf (prod.snd '' s) := rfl
+
+lemma swap_Inf [has_Inf α] [has_Inf β] (s : set (α × β)) :
+  (Inf s).swap = Inf (prod.swap '' s) :=
+ext
+  (congr_arg Inf $ image_comp prod.fst swap s : _)
+  (congr_arg Inf $ image_comp prod.snd swap s : _)
+
+lemma fst_Sup [has_Sup α] [has_Sup β] (s : set (α × β)) :
+  (Sup s).fst = Sup (prod.fst '' s) := rfl
+
+lemma snd_Sup [has_Sup α] [has_Sup β] (s : set (α × β)) :
+  (Sup s).snd = Sup (prod.snd '' s) := rfl
+
+lemma swap_Sup [has_Sup α] [has_Sup β] (s : set (α × β)) :
+  (Sup s).swap = Sup (prod.swap '' s) :=
+ext
+  (congr_arg Sup $ image_comp prod.fst swap s : _)
+  (congr_arg Sup $ image_comp prod.snd swap s : _)
+
+lemma fst_infi {ι : Sort*} (f : ι → α × β) [has_Inf α] [has_Inf β] :
```

Comment:
Can you switch to `variables {ι : Sort*} (f : ι → α × β)`?

Reply:
Turns out `ι` was already global. I've left `f` since it seems to match the rest of the file to use it locally.